### DISCUSSION TITLE
Dynamically modify the module input when dependecy is used

### DISCUSSION THREAD

**Discussion by sovyzeze** [gruntwork-io/knowledge-base#695] [How do I]
Hi all,

I want to be able to dynamically modify this terragrunt input:

ECR_URL             = dependency.ecr.outputs.ecr_url

In fact I would like to change just the ecr after dependency with different ecr names.

I tried to convert "dependency.ecr.outputs.ecr_url" in a string and manipulate it with format but it seems like the plan is returning just
"dependency.ecr.outputs.ecr_url" string and not its value.

K.R.

Ionel

---

<ins datetime="2023-04-19T10:21:54Z">
  <p><a href="https://support.gruntwork.io/hc/requests/110092">Tracked in ticket #110092</a></p>
</ins>

**Comment by denis256** [comment]
Hi,
I was wondering if can be used `replace` HCL function, depending on how much from the string should be changed, like:

```
inputs = {
    ecr_url = replace(dependency.ecr.outputs.ecr_url, "substring-to-match", "replacement-substring")
}
```

https://developer.hashicorp.com/packer/docs/templates/hcl_templates/functions/string/replace

Minimal example in: 
https://github.com/denis256/terragrunt-tests/tree/master/discussion-695

**Reply by sovyzeze** [reply]
Hi,

 

Unfortunately the replace function returns a string  the actual "dependency.<<replacement_string>>.outputs.ecr_url" whereas I need the actual ecr_url output of the dependency, for each of the dependency.

 

We are passing some vars as inputs from a terragrunt file, iterating in a list of maps.

 

xyz = { for k, v in map_list : k => {
    ECR_URL = dependency.ecr_repo_name.outputs.ecr_url
    }
}

 

ecr_repo_name needs to be replaced with the specific ecr_repo of each item. It seems you cannot use interpolation in dependency, nor can you call on a dependency in a locals block, then interpolating in the inputs.

**Reply by denis256** [reply]
hm, will be helpful to show sample code and how it is intended to work, in my [test](https://github.com/denis256/terragrunt-tests/tree/master/discussion-695) using `replace` it changed output from `087285199408.dkr.ecr.us-east-1.amazonaws.com/circle-ci-test-image-base:go1.18-tf1.4-tg39.1-pck1.8-ci50.7` to `666.dkr.ecr.n-korea.amazonaws.com/circle-ci-test-image-base:go1.18-tf1.4-tg39.1-pck1.8-ci50.7`
